code/drasil-database/lib/Drasil/Database/Chunk.hs

```diff
@@ -56,3 +55,91 @@ unChunk (Chunk c) = cast c
 -- | Ask a 'Chunk' for the type of data it codifies.
 chunkType :: Chunk -> TypeRep
 chunkType (Chunk c) = typeOf c
+
+-- | The majority of chunks will relate other chunks in some way. In other
+-- words, the majority of our chunks *depend* on others. 'HasChunkRefs' is meant
+-- as a way to capture what things a chunk *directly* refers to (i.e., depends
+-- on directly).
+class HasChunkRefs a where
+  chunkRefs :: a -> S.Set UID
+
+instance HasChunkRefs UID where
+  -- | 'UID's are meant to be "owned" (i.e., they are the unique identifier of
+  -- the chunk being defined), not *carried as references to other chunks*.
+  -- 'TypedUIDRef t' exists to be used as a *reference to another chunk of type
+  -- 't'*. Therefore, `UID` has no chunk references.
+  chunkRefs _ = S.empty
+  {-# INLINABLE chunkRefs #-}
```

Comment:
A `UID` is meant to be "owned," not carried as a reference to another chunk. However, we currently carry them as references to other chunks. We should change those references from `UID`s to `TypedUIDRef`s.